src/cargo_loading_service.cpp

```diff
@@ -12,150 +12,178 @@
-  const in_parking_msgs::msg::InParkingStatus::ConstSharedPtr msg)
+void CargoLoadingService::onTimer()
 {
-  {
-    std::lock_guard<std::mutex> lock(mutex_parking_state_);
-    aw_state_ = msg->aw_state;
+  RCLCPP_INFO(this->get_logger(), "timer start");
+  // 設備連携が完了していない
+  if (!infra_approval_) {
+    // aw_stateで条件分岐
+    switch (aw_state_) {
+      // AWがEmergencyの場合はERRORを発出し続ける
+      case InParkingStatus::AW_EMERGENCY:
+        publishCommand(static_cast<std::underlying_type<CommandState>::type>(CommandState::ERROR));
+        RCLCPP_ERROR(this->get_logger(), "AW emergency");
+        break;
+      // AWが停留所外などではSEND_ZEROを発出し、設備連携結果はFAILで返す
+      case InParkingStatus::AW_OUT_OF_PARKING:
+      case InParkingStatus::AW_UNAVAILABLE:
+        RCLCPP_WARN(this->get_logger(), "AW warning");
+        publishCommand(InfrastructureCommand::SEND_ZERO);
+        service_result_ = ExecuteInParkingTask::Response::FAIL;
+        timer_->cancel();
+        break;
+      // AWが停留所にいる場合、設備連携要求を発出
+      case InParkingStatus::AW_WAITING_FOR_ROUTE:
+      case InParkingStatus::AW_WAITING_FOR_ENGAGE:
+      case InParkingStatus::AW_ARRIVED_PARKING:
+        RCLCPP_INFO(this->get_logger(), "requesting");
+        publishCommand(
+          static_cast<std::underlying_type<CommandState>::type>(CommandState::REQUESTING));
+        break;
+      default:
+        break;
+    }
+  } else {  // 設備連携が完了
+    RCLCPP_INFO(this->get_logger(), "finished");
+    // SEND_ZEROをn秒間発出し、設備連携結果はSUCCESSで返し、timerをキャンセル
+    const auto start_time = this->now();
+    while (true) {
+      publishCommand(InfrastructureCommand::SEND_ZERO);
+      const auto time_diff = this->now() - start_time;
+      if (time_diff.seconds() > post_processing_time_) break;
+      rclcpp::sleep_for(rclcpp::Rate(command_pub_hz_).period());
+    }
+    service_result_ = ExecuteInParkingTask::Response::SUCCESS;
+    timer_->cancel();
   }
-  RCLCPP_DEBUG(this->get_logger(), "Subscribed /in_parking/state:%s",
-               rosidl_generator_traits::to_yaml(*msg).c_str());
 }
 
-void CargoLoadingService::onCargoLoadingState(
-  const v2i_interface_msgs::msg::InfrastructureStateArray::ConstSharedPtr msg)
+void CargoLoadingService::onInParkingStatus(const InParkingStatus::ConstSharedPtr msg)
 {
-  {
-    std::lock_guard<std::mutex> lock(mutex_cargo_loading_state_);
-    for (const auto & state : msg->states) {
-      if (state.id.compare(facility_info_.id) == 0) {
-        facility_info_.approval = state.approval;
-      }
-    }
+  aw_state_ = msg->aw_state;
+
+  RCLCPP_DEBUG_THROTTLE(
+    this->get_logger(), *this->get_clock(), 1000 /* ms */, "inParkingStatus: %s",
+    rosidl_generator_traits::to_yaml(*msg).c_str());
+}
+
+void CargoLoadingService::onInfrastructureStatus(const InfrastructureStateArray::ConstSharedPtr msg)
+{
+  const auto itr = std::find_if(msg->states.begin(), msg->states.end(), [this](const auto & e) {
+    return e.id == facility_id_;
+  });
```

Comment:
```suggestion
  if (infra_id_ == InfrastructureState::INVALID_ID) {
    return;
  }
  const auto itr = std::find_if(msg->states.begin(), msg->states.end(), [this](const auto & e) {
    return e.id == infra_id_;
  });
```

Reply:
```suggestion
  const auto itr = std::find_if(msg->states.begin(), msg->states.end(), [this](const auto & e) {
    return e.id == finfra_id_;
  });
```
でいいんじゃないでしょうか
INVALID_IDの条件分岐は必要ないと思います